cmd/crank/beta/trace/internal/printer/default.go

```diff
@@ -0,0 +1,192 @@
+		return errors.Wrap(err, errWriteHeader)
+	}
+
+	type queueItem struct {
+		resource *resource.Resource
+		depth    int
+		isLast   bool
+		prefix   string
+	}
+
+	// Initialize LIFO queue with root element to traverse the tree depth-first,
+	// enqueuing children in reverse order so that they are dequeued in the
+	// right order w.r.t. the way they are defined by the resources.
+	queue := []*queueItem{{resource: root}}
+
+	for len(queue) > 0 {
+		var item *queueItem
+		l := len(queue)
+		item, queue = queue[l-1], queue[:l-1] // Pop the last element
+
+		// Build the name of the current node, prepending the required prefix to
+		// show the tree structure
+		name := strings.Builder{}
+		childPrefix := item.prefix // Inherited prefix for all the children of the current node
+		switch {
+		case item.depth == 0:
+			// We don't need a prefix for the root, nor a custom
+			// prefix for its children
+		case item.isLast:
+			name.WriteString(item.prefix + "└─ ")
+			childPrefix += "   "
+		default:
+			name.WriteString(item.prefix + "├─ ")
+			childPrefix += "│  "
+		}
+
+		name.WriteString(fmt.Sprintf("%s/%s", item.resource.Unstructured.GetKind(), item.resource.Unstructured.GetName()))
+
+		// Append the namespace if it's not empty
+		if item.resource.Unstructured.GetNamespace() != "" {
+			name.WriteString(fmt.Sprintf(" (%s)", item.resource.Unstructured.GetNamespace()))
+		}
+		ready, synced, status := getResourceStatus(item.resource, p.wide)
+
+		row := defaultPrinterRow{
+			name:   name.String(),
+			ready:  ready,
+			synced: synced,
+			status: status,
+		}
+		if _, err := fmt.Fprintln(tw, row.String()); err != nil {
+			return errors.Wrap(err, errWriteRow)
+		}
+
+		// Enqueue the children of the current node in reverse order to ensure
+		// that they are dequeued from the LIFO queue in the same order w.r.t.
+		// the way they are defined by the resources.
+		for idx := len(item.resource.Children) - 1; idx >= 0; idx-- {
+			isLast := idx == len(item.resource.Children)-1
+			queue = append(queue, &queueItem{resource: item.resource.Children[idx], depth: item.depth + 1, isLast: isLast, prefix: childPrefix})
+		}
+	}
+
+	if err := tw.Flush(); err != nil {
+		return errors.Wrap(err, errFlushTabWriter)
+	}
+
+	return nil
+}
+
+// getResourceStatus returns the status of the resource.
+func getResourceStatus(r *resource.Resource, wide bool) (ready string, synced string, status string) {
+	readyCond := r.GetCondition(xpv1.TypeReady)
+	syncedCond := r.GetCondition(xpv1.TypeSynced)
+	var m string
+	switch {
+	case r.Error != nil:
+		// if there is an error we want to show it
+		status = "Error"
+		m = r.Error.Error()
+	case readyCond.Status == corev1.ConditionTrue && syncedCond.Status == corev1.ConditionTrue:
+		// if both are true we want to show the ready reason only
+		status = string(readyCond.Reason)
+
+	// The following cases are for when one of the conditions is false,
+	// prioritizing readiness over synced in case of issues.
```

Comment:
Sync comes before Ready, i.e. first the resource is synchronized, then becomes ready. So, prioritizing synced might make more sense.

Reply:
done: https://github.com/crossplane/crossplane/pull/4849/commits/f8998ceee913e53c1c4585e448f5d0936c44fb38